Source/WebCore/html/DOMURL.cpp

```diff
@@ -104,11 +104,11 @@ String DOMURL::createObjectURL(ScriptExecutionContext& scriptExecutionContext, B
 
 String DOMURL::createPublicURL(ScriptExecutionContext& scriptExecutionContext, URLRegistrable& registrable)
 {
-    URL publicURL = BlobURL::createPublicURL(scriptExecutionContext.securityOrigin());
+    auto publicURL = BlobURL::createPublicURL(scriptExecutionContext.protectedSecurityOrigin().get());
     if (publicURL.isEmpty())
-        return String();
+        return emptyString();
```

Comment:
Why? This is a behavior change.

Reply:
From the commit message:

> While here, make it return emptyString() rather than String() as that is
enforced by the bindings anyway.

Reply:
It seems slightly better if the C++ API returns the same as the JS API? I can make this `nullString()` instead if you prefer that.